frontend/app/src/components/history/TradeLocationSelector.vue

```diff
@@ -41,42 +41,55 @@
 </template>
 
 <script lang="ts">
-import { Component, Emit, Prop, Vue } from 'vue-property-decorator';
+import {
+  computed,
+  defineComponent,
+  PropType,
+  toRefs
+} from '@vue/composition-api';
+import { get } from '@vueuse/core';
 import { tradeLocations } from '@/components/history/consts';
 import LocationIcon from '@/components/history/LocationIcon.vue';
 import { TradeLocationData } from '@/components/history/type';
 import { TradeLocation } from '@/services/history/types';
 
-@Component({
-  components: { LocationIcon }
-})
-export default class TradeLocationSelector extends Vue {
-  readonly tradeLocations = tradeLocations;
+export default defineComponent({
+  name: 'TradeLocationSelector',
+  components: { LocationIcon },
+  props: {
+    value: { required: true, type: Object as PropType<TradeLocation> },
+    availableLocations: { required: false, type: Array, default: () => [] },
```

Comment:
is available locations a `string[]` maybe we can also properly type it with `as PropType<...>`